client/src/client.rs

```diff
@@ -986,46 +1172,56 @@ pub trait RpcApi: Sized {
             pub start_height: usize,
             pub stop_height: Option<usize>,
         }
-        let res: Response =
-            self.call("rescanblockchain", handle_defaults(&mut args, &[0.into(), null()]))?;
+        let res: Response = self
+            .call(
+                "rescanblockchain",
+                handle_defaults(&mut args, &[0.into(), null()]),
+            )
+            .await?;
         Ok((res.start_height, res.stop_height))
     }
 
     /// Returns statistics about the unspent transaction output set.
     /// This call may take some time.
-    fn get_tx_out_set_info(&self) -> Result<json::GetTxOutSetInfoResult> {
-        self.call("gettxoutsetinfo", &[])
+    async fn get_tx_out_set_info(&self) -> Result<json::GetTxOutSetInfoResult> {
+        self.call("gettxoutsetinfo", &[]).await
     }
 
     /// Returns information about network traffic, including bytes in, bytes out,
     /// and current time.
-    fn get_net_totals(&self) -> Result<json::GetNetTotalsResult> {
-        self.call("getnettotals", &[])
+    async fn get_net_totals(&self) -> Result<json::GetNetTotalsResult> {
+        self.call("getnettotals", &[]).await
     }
 
     /// Returns the estimated network hashes per second based on the last n blocks.
-    fn get_network_hash_ps(&self, nblocks: Option<u64>, height: Option<u64>) -> Result<f64> {
+    async fn get_network_hash_ps(&self, nblocks: Option<u64>, height: Option<u64>) -> Result<f64> {
         let mut args = [opt_into_json(nblocks)?, opt_into_json(height)?];
-        self.call("getnetworkhashps", handle_defaults(&mut args, &[null(), null()]))
+        self.call(
+            "getnetworkhashps",
+            handle_defaults(&mut args, &[null(), null()]),
+        )
+        .await
     }
 
     /// Returns the total uptime of the server in seconds
-    fn uptime(&self) -> Result<u64> {
-        self.call("uptime", &[])
+    async fn uptime(&self) -> Result<u64> {
+        self.call("uptime", &[]).await
     }
 }
 
 /// Client implements a JSON-RPC client for the Bitcoin Core daemon or compatible APIs.
 pub struct Client {
-    client: jsonrpc::client::Client,
+    url: String,
+    client: reqwest::Client,
+    nonce: Arc<Mutex<u64>>,
```

Comment:
Citing the documentation for `tokio::sync::Mutex`:

> Contrary to popular belief, it is ok and often preferred to use the ordinary Mutex from the standard library in asynchronous code. This section will help you decide on which kind of mutex you should use.
> 
> The primary use case of the async mutex is to provide shared mutable access to IO resources such as a database connection. If the data stored behind the mutex is just data, it is often better to use a blocking mutex such as the one in the standard library or parking_lot. This is because the feature that the async mutex offers over the blocking mutex is that it is possible to keep the mutex locked across an .await point, which is rarely necessary for data.

So I think here we can use a standard Mutex and drop the tokio dependency.